cloud/amazon/lambda.py

```diff
@@ -0,0 +1,394 @@
+      - The runtime environment for the Lambda function you are uploading. Required when creating a function.
+    default: null
+    choices: [ 'nodejs', 'java8', 'python2.7' ]
+  role_arn:
+    description:
+      - The Amazon Resource Name (ARN) of the IAM role that Lambda assumes when it executes your function to access any other Amazon Web Services (AWS) resources
+    default: null
+  handler:
+    description:
+      - The function within your code that Lambda calls to begin execution
+    default: null
+  path:
+    description:
+      - A .zip file containing your deployment package
+    required: false
+    default: null
+    aliases: [ 'src' ]
+  s3_bucket:
+    description:
+      - Amazon S3 bucket name where the .zip file containing your deployment package is stored
+    required: false
+    default: null
+  s3_key:
+    description:
+      - The Amazon S3 object (the deployment package) key name you want to upload
+    required: false
+    default: null
+  s3_object_version:
+    description:
+      - The Amazon S3 object (the deployment package) version you want to upload.
+    required: false
+    default: null
+  description:
+    description:
+      - A short, user-defined function description. Lambda does not use this value. Assign a meaningful description as you see fit.
+    required: false
+    default: null
+  timeout:
+    description:
+      - The function execution time at which Lambda should terminate the function.
+    required: false
+    default: 3
+  memory_size:
+    description:
+      - The amount of memory, in MB, your Lambda function is given
+    required: false
+    default: 128
+notes:
+  - 'Currently this module only supports uploaded code via S3'
+author:
+    - 'Steyn Huizinga (@steynovich)'
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Create Lambda functions
+tasks:
+- name: looped creation
+  lambda:
+    name: '{{ item.name }}'
+    state: present
+    path: '{{ item.path }}'
+    runtime: 'python2.7'
+    role_arn: 'arn:aws:iam::987654321012:role/lambda_basic_execution'
+    handler: 'hello_python.my_handler'
+  with_items:
+    - { name: HelloWorld, path: 'hello-code.zip' }
+    - { name: ByeBye, path: 'bye-code.zip' }
+
+# Basic Lambda function deletion
+tasks:
+- name: Delete Lambda functions HelloWorld and ByeBye
+  lambda:
+    name: '{{ item }}'
+    state: absent
+  with_items:
+    - HelloWorld
+    - ByeBye
+'''
+
+RETURN = '''
+output:
+  description: the data returned by create_function in boto3
+  returned: success
+  type: dict
+  sample: 
+    {
+        'FunctionName': 'string',
+        'FunctionArn': 'string',
+        'Runtime': 'nodejs',
+        'Role': 'string',
+        'Handler': 'string',
+        'CodeSize': 123,
+        'Description': 'string',
+        'Timeout': 123,
+        'MemorySize': 123,
+        'LastModified': 'string',
+        'CodeSha256': 'string',
+        'Version': 'string',
+    }
+'''
+
+try:
+    import boto
```

Comment:
I can't see where boto is used in your code (and it shouldn't be as new modules must be boto3 only) so is this necessary?